[2024-12-26T17:25:52.606+0000] {local_task_job_runner.py:123} INFO - ::group::Pre task execution logs
[2024-12-26T17:25:53.883+0000] {taskinstance.py:2613} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: movie_pipeline.parse_collection_page manual__2024-12-26T17:07:20.299024+00:00 [queued]>
[2024-12-26T17:25:53.935+0000] {taskinstance.py:2613} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: movie_pipeline.parse_collection_page manual__2024-12-26T17:07:20.299024+00:00 [queued]>
[2024-12-26T17:25:53.938+0000] {taskinstance.py:2866} INFO - Starting attempt 4 of 4
[2024-12-26T17:25:54.104+0000] {taskinstance.py:2889} INFO - Executing <Task(PythonOperator): parse_collection_page> on 2024-12-26 17:07:20.299024+00:00
[2024-12-26T17:25:54.123+0000] {standard_task_runner.py:72} INFO - Started process 4482 to run task
[2024-12-26T17:25:54.185+0000] {standard_task_runner.py:104} INFO - Running: ['airflow', 'tasks', 'run', 'movie_pipeline', 'parse_collection_page', 'manual__2024-12-26T17:07:20.299024+00:00', '--job-id', '50', '--raw', '--subdir', 'DAGS_FOLDER/pipeline_dag.py', '--cfg-path', '/tmp/tmplraegp1_']
[2024-12-26T17:25:54.257+0000] {standard_task_runner.py:105} INFO - Job 50: Subtask parse_collection_page
[2024-12-26T17:25:54.662+0000] {task_command.py:467} INFO - Running <TaskInstance: movie_pipeline.parse_collection_page manual__2024-12-26T17:07:20.299024+00:00 [running]> on host 1712831d7ea4
[2024-12-26T17:25:55.102+0000] {abstractoperator.py:783} ERROR - Exception rendering Jinja template for task 'parse_collection_page', field 'op_args'. Template: ["{{ti.xcom_pull(key='return_value',task_ids='scrape_collection_page')}}", '{{(logical_datemacros.timedelta(days=2)).date()}}']
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/models/abstractoperator.py", line 775, in _do_render_template_fields
    rendered_content = self.render_template(
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/template/templater.py", line 183, in render_template
    return [self.render_template(element, context, jinja_env, oids) for element in value]
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/template/templater.py", line 183, in <listcomp>
    return [self.render_template(element, context, jinja_env, oids) for element in value]
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/template/templater.py", line 171, in render_template
    return self._render(template, context)
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/models/abstractoperator.py", line 730, in _render
    return super()._render(template, context, dag=dag)
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/template/templater.py", line 127, in _render
    return render_template_to_string(template, context)
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/utils/helpers.py", line 301, in render_template_to_string
    return render_template(template, cast(MutableMapping[str, Any], context), native=False)
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/utils/helpers.py", line 296, in render_template
    return "".join(nodes)
  File "<template>", line 12, in root
  File "/home/airflow/.local/lib/python3.9/site-packages/jinja2/sandbox.py", line 327, in getattr
    value = getattr(obj, attribute)
  File "/home/airflow/.local/lib/python3.9/site-packages/jinja2/runtime.py", line 864, in __getattr__
    return self._fail_with_undefined_error()
  File "/home/airflow/.local/lib/python3.9/site-packages/jinja2/runtime.py", line 857, in _fail_with_undefined_error
    raise self._undefined_exception(self._undefined_message)
jinja2.exceptions.UndefinedError: 'logical_datemacros' is undefined
[2024-12-26T17:25:55.121+0000] {taskinstance.py:3311} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/models/taskinstance.py", line 273, in _run_raw_task
    TaskInstance._execute_task_with_callbacks(
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/models/taskinstance.py", line 3115, in _execute_task_with_callbacks
    task_orig = self.render_templates(context=context, jinja_env=jinja_env)
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/models/taskinstance.py", line 3534, in render_templates
    original_task.render_template_fields(context, jinja_env)
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/models/baseoperator.py", line 1438, in render_template_fields
    self._do_render_template_fields(self, self.template_fields, context, jinja_env, set())
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/models/abstractoperator.py", line 775, in _do_render_template_fields
    rendered_content = self.render_template(
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/template/templater.py", line 183, in render_template
    return [self.render_template(element, context, jinja_env, oids) for element in value]
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/template/templater.py", line 183, in <listcomp>
    return [self.render_template(element, context, jinja_env, oids) for element in value]
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/template/templater.py", line 171, in render_template
    return self._render(template, context)
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/models/abstractoperator.py", line 730, in _render
    return super()._render(template, context, dag=dag)
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/template/templater.py", line 127, in _render
    return render_template_to_string(template, context)
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/utils/helpers.py", line 301, in render_template_to_string
    return render_template(template, cast(MutableMapping[str, Any], context), native=False)
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/utils/helpers.py", line 296, in render_template
    return "".join(nodes)
  File "<template>", line 12, in root
  File "/home/airflow/.local/lib/python3.9/site-packages/jinja2/sandbox.py", line 327, in getattr
    value = getattr(obj, attribute)
  File "/home/airflow/.local/lib/python3.9/site-packages/jinja2/runtime.py", line 864, in __getattr__
    return self._fail_with_undefined_error()
  File "/home/airflow/.local/lib/python3.9/site-packages/jinja2/runtime.py", line 857, in _fail_with_undefined_error
    raise self._undefined_exception(self._undefined_message)
jinja2.exceptions.UndefinedError: 'logical_datemacros' is undefined
[2024-12-26T17:25:55.156+0000] {taskinstance.py:1225} INFO - Marking task as FAILED. dag_id=movie_pipeline, task_id=parse_collection_page, run_id=manual__2024-12-26T17:07:20.299024+00:00, execution_date=20241226T170720, start_date=20241226T172553, end_date=20241226T172555
[2024-12-26T17:25:55.190+0000] {taskinstance.py:340} INFO - ::group::Post task execution logs
[2024-12-26T17:25:55.193+0000] {standard_task_runner.py:124} ERROR - Failed to execute job 50 for task parse_collection_page ('logical_datemacros' is undefined; 4482)
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/task/task_runner/standard_task_runner.py", line 117, in _start_by_fork
    ret = args.func(args, dag=self.dag)
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/cli/cli_config.py", line 49, in command
    return func(*args, **kwargs)
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/utils/cli.py", line 116, in wrapper
    return f(*args, **kwargs)
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/cli/commands/task_command.py", line 483, in task_run
    task_return_code = _run_task_by_selected_method(args, _dag, ti)
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/cli/commands/task_command.py", line 256, in _run_task_by_selected_method
    return _run_raw_task(args, ti)
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/cli/commands/task_command.py", line 341, in _run_raw_task
    return ti._run_raw_task(
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/utils/session.py", line 97, in wrapper
    return func(*args, session=session, **kwargs)
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/models/taskinstance.py", line 3005, in _run_raw_task
    return _run_raw_task(
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/models/taskinstance.py", line 273, in _run_raw_task
    TaskInstance._execute_task_with_callbacks(
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/models/taskinstance.py", line 3115, in _execute_task_with_callbacks
    task_orig = self.render_templates(context=context, jinja_env=jinja_env)
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/models/taskinstance.py", line 3534, in render_templates
    original_task.render_template_fields(context, jinja_env)
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/models/baseoperator.py", line 1438, in render_template_fields
    self._do_render_template_fields(self, self.template_fields, context, jinja_env, set())
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/models/abstractoperator.py", line 775, in _do_render_template_fields
    rendered_content = self.render_template(
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/template/templater.py", line 183, in render_template
    return [self.render_template(element, context, jinja_env, oids) for element in value]
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/template/templater.py", line 183, in <listcomp>
    return [self.render_template(element, context, jinja_env, oids) for element in value]
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/template/templater.py", line 171, in render_template
    return self._render(template, context)
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/models/abstractoperator.py", line 730, in _render
    return super()._render(template, context, dag=dag)
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/template/templater.py", line 127, in _render
    return render_template_to_string(template, context)
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/utils/helpers.py", line 301, in render_template_to_string
    return render_template(template, cast(MutableMapping[str, Any], context), native=False)
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/utils/helpers.py", line 296, in render_template
    return "".join(nodes)
  File "<template>", line 12, in root
  File "/home/airflow/.local/lib/python3.9/site-packages/jinja2/sandbox.py", line 327, in getattr
    value = getattr(obj, attribute)
  File "/home/airflow/.local/lib/python3.9/site-packages/jinja2/runtime.py", line 864, in __getattr__
    return self._fail_with_undefined_error()
  File "/home/airflow/.local/lib/python3.9/site-packages/jinja2/runtime.py", line 857, in _fail_with_undefined_error
    raise self._undefined_exception(self._undefined_message)
jinja2.exceptions.UndefinedError: 'logical_datemacros' is undefined
[2024-12-26T17:25:55.249+0000] {local_task_job_runner.py:266} INFO - Task exited with return code 1
[2024-12-26T17:25:55.331+0000] {taskinstance.py:3895} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2024-12-26T17:25:55.353+0000] {local_task_job_runner.py:245} INFO - ::endgroup::
